### DISCUSSION TITLE
How to use with actix-web

### DISCUSSION THREAD

**Discussion by DarrenBaldwin07** [Totodore/socketioxide#207] [Q&A] [answered]
- Would be interested to see how this could work with actix-web?
- I know its not hyper/tower based so could potentially be difficult?

**Comment by Totodore** [comment, marked as answer]
It would require the refactoring of *a lot* of things and the introduction of separated modules. A lot of things are currently based on tokio :
* New tasks 
* Timers
* Channels
* Tokio-tungstenite

For the moment it is absolutely not planned because the codebase relies too deeply on tokio/tower/hyper. Also there are currently a lot of things to do to get something more stable and production ready. And working on an actix integration would basically be going in the opposite direction.

**Comment by DarrenBaldwin07** [comment]
Totally get it - thanks!
